### ISSUE TITLE
PIO version check fails with PGI 18.10 compiler

### ISSUE THREAD

**Issue by mgduda** [MPAS-Dev/MPAS-Model#207]
The check in the top-level `Makefile` for a working PIO library fails with the PGI 18.10 compiler, apparently because the source files need to be specified at the beginning of the compiler's command-line arguments.

As-is, we get the following error (for what should be a valid PIO2 library):
```
bash-4.2$ mpif90  -I/glade/work/duda/pio-2.3.1-pgi18.10/include -I/glade/work/duda/libs-pgi18.10-skylake/include -I/glade/work/duda/libs-pgi18.10-skylake/include -I/glade/work/duda/libs-pgi18.10-skylake/include -O3 -byteswapio -Mfree -O3 -L/glade/work/duda/pio-2.3.1-pgi18.10/lib -lpiof -lpioc -L/glade/work/duda/libs-pgi18.10-skylake/lib -lnetcdff -lnetcdf -L/glade/work/duda/libs-pgi18.10-skylake/lib -lpnetcdf -L/glade/work/duda/libs-pgi18.10-skylake/lib -lhdf5_hl -lhdf5 -ldl -lz -o pio2.out pio2.f90
/glade/scratch/duda/pgf90vUZc-XbvgynC.o: In function `.C1_283':
pio2.f90:(.data+0x4): undefined reference to `pio_'
pio2.f90:(.data+0x1c): undefined reference to `pio_support_'
pio2.f90:(.data+0x2c): undefined reference to `pio_kinds_'
pio2.f90:(.data+0x34): undefined reference to `pio_types_'
pio2.f90:(.data+0x3c): undefined reference to `pionfget_mod_'
pio2.f90:(.data+0x44): undefined reference to `pionfput_mod_'
pio2.f90:(.data+0x4c): undefined reference to `pionfatt_mod_'
pio2.f90:(.data+0x54): undefined reference to `pio_nf_'
pio2.f90:(.data+0x5c): undefined reference to `piodarray_'
pio2.f90:(.data+0x64): undefined reference to `piolib_mod_'
```

Moving the `pio2.f90` argument to the beginning of the compiler arguments seems to resolve the issue:
```
bash-4.2$ mpif90 pio2.f90 -I/glade/work/duda/pio-2.3.1-pgi18.10/include -I/glade/work/duda/libs-pgi18.10-skylake/include -I/glade/work/duda/libs-pgi18.10-skylake/include -I/glade/work/duda/libs-pgi18.10-skylake/include -O3 -byteswapio -Mfree -O3 -L/glade/work/duda/pio-2.3.1-pgi18.10/lib -lpiof -lpioc -L/glade/work/duda/libs-pgi18.10-skylake/lib -lnetcdff -lnetcdf -L/glade/work/duda/libs-pgi18.10-skylake/lib -lpnetcdf -L/glade/work/duda/libs-pgi18.10-skylake/lib -lhdf5_hl -lhdf5 -ldl -lz -o pio2.out
```

**Comment by mgduda** [comment]
@MiCurry If you'd like to do some testing with the PGI 18.10 compilers on Casper, you can start an interactive session with `execdav -C skylake`, then source `/glade/work/duda/libs-pgi18.10-skylake/setup` (assuming the bash shell).

**Comment by MiCurry** [comment]
@mgduda see PR #212.

**Comment by mgduda** [comment]
Resolved by PR #212 .
